### ISSUE TITLE
Timeout on stop/start after daemon restart

### ISSUE THREAD

**Issue by ricab** [canonical/multipass#1243]
1. `multipass start`
2. `snap restart multipass`
3. `multipass stop`
4. `multipass start`
=> Timeout

Daemon's log says `qemu-system-x86_64: Device 'hda' does not have the requested snapshot 'suspend'` when resuming from stopped (not when resuming from the daemon restart).

**Comment by Saviq** [comment]
Confirmed, we're now deleting the snapshot properly (#725), but not resetting the state of the machine to know it's not suspended.

**Comment by Saviq** [comment]
To recover (there's no risk of data loss):
```
multipass stop <instance>
snap restart multipass
multipass start <instance>
```

**Comment by townsend2010** [comment]
Ah, right, it's not the state that's wrong, it's the saved `qemu` options that are not updated when the snapshot is deleted.  This has probably been a problem since day 1 of supporting `suspend` :grin:

**Comment by townsend2010** [comment]
Actually, on a second look, I think this regressed when we switched to the `QemuVMProcessSpec`.

**Comment by ricab** [comment]
@townsend2010 I am not sure how useful it is, but I could only get this with a daemon restart. When I issued suspend + start instead, I would not see the issue.

**Comment by townsend2010** [comment]
Ugh, not only this issue, but I discovered an even worse issue in debugging this.  Doing the start + suspend + start, I found that the start after the suspend is not loading the saved memory state, so it's not working at all.  When we switched to using the `QemuVMProcessSpec`, it sets the `qemu-system-x86_64` arguments *only* in `QemuVirtualMachine`'s ctor and never accounts for any arguments changes after this.  However, the `-loadvm` option needs to be added/removed dynamically depending on the new state of the VM like if it's coming out of suspend or not.
